Added VTOL check for airspeed increase during front transition

RomanBapst opened the pull request:
### Solved Problem
When the airspeed sensor is blocked, a VTOL will remain in the front transition until the transition times out (duration defiend by parameter). During this time the vehicle can speed up considerably and the rate controller can go into a limit cycle since the airspeed scaling does not work properly anymore.
Furthermore, the airspeed validator does not check against airspeed failures during the transition.

### Solution
If the vehicle uses airspeed, then we can be more strict on the transition timeout logic, by requiring that the airspeed should have reached at least the blending airspeed after the open loop transition time. The open loop transition time is usually set to the expected transition time and the airspeed blend parameter is usually well below the transition airspeed. Therefore, this condition should be quite robust and should not cause any false positives when the system is working as expected.

### Changelog Entry
For release notes:
```
Feature: For VTOL front transition with an airspeed sensor, require airspeed to reach blend airspeed after openloop transition time, otherwhise transition timeout will be triggered.
New parameter: XYZ_Z
Documentation: Add section in VTOL tuning guide which explains the change.
```

### Test coverage
Tested on standard VTOL and in SITL.
### Context
Related links, screenshot before/after, video

sfuhrer commented on src/modules/airspeed_selector/AirspeedValidator.cpp:
```diff
@@ -339,11 +351,21 @@ AirspeedValidator::check_first_principle(const uint64_t timestamp, const float t
 	}
 }
 
+void AirspeedValidator::checkAirspeedIncreaseDuringFrontTransition(const uint64_t timestamp,
```
Like the other airspeed checks this one should get a bit in ASPD_DO_CHECKS to enable/disable it.

sfuhrer reviewed (commented):
With the proposed method we cannot distinguish between airspeed sensing or fixed-wing actuation problems, can we? 
If we can't then we need to adapt the message sent to the user (airspeed OR FW actuation problem detected, abort transition).
How does this check differ fundamentally from the one around VT_TRANS_TIMEOUT?

RomanBapst commented:
@sfuhrer 

> With the proposed method we cannot distinguish between airspeed sensing or fixed-wing actuation problems, can we?

No, for that we would need to add groundspeed or acceleration information. But it's a good point. If we detect this failure then we might also want to disable pusher assist...


>  How does this check differ fundamentally from the one around VT_TRANS_TIMEOUT?

The transition timeout is e.g. used when we don't have an airspeed sensor.

RomanBapst commented:
As discussed with @sfuhrer, I removed the airspeed failure detection in the airspeed validator. If airspeed does not increase during the transition it does not necessarily need to be due to a faulty airspeed sensor, as it could also be an actuation issue.

Instead, we use the existing transition timeout logic for quadchute. If airspeed is enabled, we can timeout the transition earlier, as we can require the airspeed to have reached at least the blend airspeed after the openloop front transition time.

I updated the parameter description for `VT_F_TR_OL_TM` and will add this parameter to the documentation. It's important to mention that the parameter value also has a meaning now if you have airspeed feedback available.

sfuhrer commented on src/modules/vtol_att_control/vtol_type.cpp:
```diff
@@ -329,10 +329,13 @@ bool VtolType::isRollExceeded()
 
-		if (_time_since_trans_start > getFrontTransitionTimeout()) {
-			// transition timeout occured, abort transition
+		} else if (_time_since_trans_start > getFrontTransitionTimeout()) {
```
```suggestion
		} else if (getFrontTransitionTimeout() > FLT_EPSILON && _time_since_trans_start > getFrontTransitionTimeout()) {
```

RomanBapst replied:
@sfuhrer The transition timeout should according to the param docs always be positive. I also think that one should always have a transition timeout enabled.
I adjusted the param documentation to remove the part which says that you can disable it.
